apis/Google.Cloud.Firestore/Google.Cloud.Firestore/FirestoreDeserializationConfigurationAttribute.cs

```diff
@@ -53,12 +53,12 @@ public sealed class FirestoreDeserializationConfigurationAttribute : Attribute
         /// </summary>
         /// <param name="context">The current deserialization context.</param>
         /// <param name="map">The map to (possibly) update.</param>
-        internal void ApplyContext(DeserializationContext context, Dictionary<string, object> map)
+        internal void ApplyContext(IDeserializationContext context, Dictionary<string, object> map)
         {
             MaybePopulate(DocumentIdKey, context.DocumentReference);
-            MaybePopulate(CreateTimestampKey, context.Snapshot.CreateTime);
-            MaybePopulate(UpdateTimestampKey, context.Snapshot.UpdateTime);
-            MaybePopulate(ReadTimestampKey, context.Snapshot.ReadTime);
+            MaybePopulate(CreateTimestampKey, context.CreateTime);
```

Comment:
I think this needs to be something like:
```suggestion
            MaybePopulate(CreateTimestampKey, () => context.CreateTime);
```

And then the function below:

```
void MaybePopulate(string key, Func<object> valueGetter)
{
    ...
}
```

to defer attempting to obtain the value until we know the key is not null. Else this will always fail even if the attribute is not configured to deserialize the non-general values?

But I might be missing something here.

Reply:
Ooh, you're absolutely right. Will write a test first, then fix it.

Reply:
Done now in a second commit - PTAL.